exportc symbol not exported, leading to link error

`exportc` symbol is not exported in some cases, leading to link error. This seems to occur only when the proc body is disabled via conditional compilation (eg via `when false_condition: foo` or similar).

### Example
```nim
# fun1.nim
proc myfun() {.importc.}
proc myfunWrap*() = myfun()
```

```nim
# main.nim
import ./fun1

proc myfun() {.exportc.} = # ditto with {.exportc, dynlib.}
  # discard # works
  when false: discard # has the BUG

proc main() = myfunWrap()
main()
```

### Current Output
nim c main.nim
```
Undefined symbols for architecture x86_64:
  "_myfun", referenced from:
      _myfunWrap_i40p9axSiEKMrYIQZ70IbKw in fun1.nim.c.o
```

### Expected Output
should compile and link

### Possible Solution
looks like when the body is disabled (eg via `when false: foo`) the exportc symbol is not exported; it works with an empty `discard`; looks like a codegen bug

### Additional Information
* Your Nim version (output of `nim -v`).
tried 0.18 through 0.20 + latest devel 0718d6c2cddac89256e2fcde17407f1685f1a457; all versions have that issue
(on osx but shouldn't matter)

* A link to a related issue or discussion.

I use this pattern (importc+exportc) as a workaround for lack of cyclic imports (as discussed here https://forum.nim-lang.org/t/4004#24925), which allows me to reduce dependencies between certain key modules; actually the pattern I use is a bit more sophisticated and allows to also skip type definitions using some macro magic. The reduced bug above is what I had encountered here: https://forum.nim-lang.org/t/4004#24925 (`how to forward declare a proc in module A and define it in module B?`)

I currently have to work around this bug by adding dummy calls to the exported procs.

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -2669,9 +2669,8 @@ proc expr(p: BProc, n: PNode, d: var TLoc) =
         if ({sfExportc, sfCompilerProc} * prc.flags == {sfExportc}) or
             (sfExportc in prc.flags and lfExportLib in prc.loc.flags) or
             (prc.kind == skMethod):
-          # we have not only the header:
-          if prc.getBody.kind != nkEmpty or lfDynamicLib in prc.loc.flags:
-            genProc(p.module, prc)
+          # Generate proc even if empty body, bugfix #11651.
+          genProc(p.module, prc)
   of nkParForStmt: genParForStmt(p, n)
   of nkState: genState(p, n)
   of nkGotoState:

